src/lib/y2firewall/dialogs/main.rb

```diff
@@ -130,6 +130,17 @@ def swap_api
       # modifications
       def apply_changes
         return false if Yast::Mode.config
+        # Firewall settings will be written into the permanent configurations only.
+        # So the running firewalld service will not be changed. Even a reload does
+        # not help (see man pages). So the running firewalld service has to be
+        # restarted.
+        # Set a flag only. Restarting will be done by system_service.save.
+        if fw.modified? && # Data has been changed by user
+            fw.system_service.running? &&      # The service is already running
+            fw.system_service.action != :stop  # and will not be stopped by the user
```

Comment:
Still not sure at all, I mean, if the user select a reload after writing we will restart the service that isn't what was requested at all.

Reply:
I see your point. Well, from my point of view I would not offer "restart" or "reload" at all to the user, because even for us it is not really clear in all corner cases. So I would say...keep it simple for the user. What do you think about it ?

Reply:
Yep, that are the consequences of moving to the service_widget, it is a general approach while in the past it was handled by the firewall itself.

BTW, the only problematic case is when the service is `running` and the configuration has been modified, in this case a firewalld --reload should be enough what is basically what `fw.write` does.

So maybe something like

```ruby
... 
  return fw.write if need_reload?

  fw.write_only
  fw.system_service.save
end

def need_reload?
  fw.modified? && fw.system_service.running? && !fw.system_service.action
end

```

Reply:
@schubi2 

**Note:** this is the bug about --reload that I was talking about https://bugzilla.opensuse.org/show_bug.cgi?id=1112008